In [1]:
import pandas as pd


In [2]:
msa2000_1 = pd.read_csv('../data/OEWS/2000msa_dl_1.csv')
msa2000_1.head(2)

prim_state  area          area_name occ_code                occ_titl  group  \
0         AK   380  Anchorage, AK MSA  11-0000  Management Occupations  major   
1         AK   380  Anchorage, AK MSA  11-1011        Chief Executives    NaN   

        tot_emp emp_prse   h_mean     a_mean  ... h_wpct75 h_wpct90 a_wpct10  \
0         10410      4.0    32.84      68310  ...    41.76    62.45    31440   
1           480     10.4    51.54     107200  ...        #        #    65340   

  a_wpct25 a_median a_wpct75 a_wpct90 annual release  year  
0    42780    61900    86860   129900    NaN     NaN  2000  
1    80460   109270        #        #    NaN     NaN  2000  

[2 rows x 24 columns]

In [3]:
msa2000_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60472 entries, 0 to 60471
Data columns (total 24 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   prim_state  60472 non-null  object 
 1   area        60472 non-null  int64  
 2   area_name   60472 non-null  object 
 3   occ_code    60472 non-null  object 
 4   occ_titl    60467 non-null  object 
 5   group       3643 non-null   object 
 6   tot_emp     60472 non-null  object 
 7   emp_prse    60472 non-null  object 
 8   h_mean      60472 non-null  object 
 9   a_mean      60472 non-null  object 
 10  mean_prse   60472 non-null  object 
 11  h_wpct10    60472 non-null  object 
 12  h_wpct25    60472 non-null  object 
 13  h_median    60472 non-null  object 
 14  h_wpct75    60472 non-null  object 
 15  h_wpct90    60472 non-null  object 
 16  a_wpct10    60472 non-null  object 
 17  a_wpct25    60472 non-null  object 
 18  a_median    60472 non-null  object 
 19  a_wpct75    60472 non-nul

In [4]:
msa2000_2 = pd.read_csv('../data/OEWS/2000msa_dl_2.csv')
msa2000_2.head(2)

prim_state  area                           area_name occ_code  \
0         MS   920  Biloxi-Gulfport-Pascagoula, MS MSA  11-0000   
1         MS   920  Biloxi-Gulfport-Pascagoula, MS MSA  11-1011   

                 occ_titl  group       tot_emp emp_prse   h_mean     a_mean  \
0  Management Occupations  major          8060     12.3    26.37      54860   
1        Chief Executives    NaN           400     20.7    45.39      94410   

   ... h_wpct75 h_wpct90 a_wpct10 a_wpct25 a_median a_wpct75 a_wpct90 annual  \
0  ...    34.00    45.48    22920    32280    49150    70710    94590    NaN   
1  ...        #        #    36380    56260    89510        #        #    NaN   

  release  year  
0     NaN  2000  
1     NaN  2000  

[2 rows x 24 columns]

In [5]:
msa2000_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60631 entries, 0 to 60630
Data columns (total 24 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   prim_state  60631 non-null  object 
 1   area        60631 non-null  int64  
 2   area_name   60631 non-null  object 
 3   occ_code    60631 non-null  object 
 4   occ_titl    60628 non-null  object 
 5   group       3673 non-null   object 
 6   tot_emp     60631 non-null  object 
 7   emp_prse    60631 non-null  object 
 8   h_mean      60631 non-null  object 
 9   a_mean      60631 non-null  object 
 10  mean_prse   60631 non-null  object 
 11  h_wpct10    60631 non-null  object 
 12  h_wpct25    60631 non-null  object 
 13  h_median    60631 non-null  object 
 14  h_wpct75    60631 non-null  object 
 15  h_wpct90    60631 non-null  object 
 16  a_wpct10    60631 non-null  object 
 17  a_wpct25    60631 non-null  object 
 18  a_median    60631 non-null  object 
 19  a_wpct75    60631 non-nul

### Append the dataframes

In [6]:
commsa2000 = msa2000_1.append(msa2000_2, ignore_index=True, verify_integrity=True, sort=False)
commsa2000.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 121103 entries, 0 to 121102
Data columns (total 24 columns):
 #   Column      Non-Null Count   Dtype  
---  ------      --------------   -----  
 0   prim_state  121103 non-null  object 
 1   area        121103 non-null  int64  
 2   area_name   121103 non-null  object 
 3   occ_code    121103 non-null  object 
 4   occ_titl    121095 non-null  object 
 5   group       7316 non-null    object 
 6   tot_emp     121103 non-null  object 
 7   emp_prse    121103 non-null  object 
 8   h_mean      121103 non-null  object 
 9   a_mean      121103 non-null  object 
 10  mean_prse   121103 non-null  object 
 11  h_wpct10    121103 non-null  object 
 12  h_wpct25    121103 non-null  object 
 13  h_median    121103 non-null  object 
 14  h_wpct75    121103 non-null  object 
 15  h_wpct90    121103 non-null  object 
 16  a_wpct10    121103 non-null  object 
 17  a_wpct25    121103 non-null  object 
 18  a_median    121103 non-null  object 
 19  a_

### Keep only necessary columns & rename columns to match 2010/2017 column names.

In [7]:
commsa2000 = commsa2000[['prim_state', 'area', 'area_name', 'occ_code', 'occ_titl', 'tot_emp', 'a_mean', 'a_wpct10', 
                         'a_wpct25', 'a_median', 'a_wpct75', 'a_wpct90', 'year']]

commsa2000 = commsa2000.rename(columns= {'occ_titl': 'occ_title','a_wpct10':'a_pct10', 'a_wpct25' : 'a_pct25', 
                                         'a_wpct75': 'a_pct75', 'a_wpct90': 'a_pct90'})
commsa2000.head(2)

prim_state  area          area_name occ_code               occ_title  \
0         AK   380  Anchorage, AK MSA  11-0000  Management Occupations   
1         AK   380  Anchorage, AK MSA  11-1011        Chief Executives   

        tot_emp     a_mean a_pct10 a_pct25 a_median a_pct75 a_pct90  year  
0         10410      68310   31440   42780    61900   86860  129900  2000  
1           480     107200   65340   80460   109270       #       #  2000

### Keep only relevant OCC codes for the year 2000.

In [8]:
occ_codes_2000 = ['11-3021', '13-1111', '15-1011', '15-1021', '15-1031',                                                      
                       '15-1032', '15-1041', '15-1051', '15-1061', '15-1071', 
                       '15-1081', '15-2031', '15-2041', '17-2061','19-3021',
                       '43-9011', '43-9021', '43-9111', '49-2011']
commsa2000 = commsa2000.loc[commsa2000.occ_code.isin(occ_codes_2000)].reset_index(drop=True)
commsa2000


prim_state  area          area_name occ_code  \
0            AK   380  Anchorage, AK MSA  11-3021   
1            AK   380  Anchorage, AK MSA  13-1111   
2            AK   380  Anchorage, AK MSA  15-1021   
3            AK   380  Anchorage, AK MSA  15-1031   
4            AK   380  Anchorage, AK MSA  15-1041   
...         ...   ...                ...      ...   
4346         WY  1580   Cheyenne, WY MSA  15-1051   
4347         WY  1580   Cheyenne, WY MSA  15-1071   
4348         WY  1580   Cheyenne, WY MSA  43-9011   
4349         WY  1580   Cheyenne, WY MSA  43-9021   
4350         WY  1580   Cheyenne, WY MSA  49-2011   

                                              occ_title       tot_emp  \
0             Computer and Information Systems Managers           210   
1                                   Management Analysts           180   
2                                  Computer Programmers           380   
3             Computer Software Engineers, Applications            50   
4                          Computer Support Specialists           720   
...                                                 ...           ...   
4346                          Computer Systems Analysts            70   
4347        Network and Computer Systems Administrators            40   
4348                                 Computer Operators            80   
4349                                  Data Entry Keyers           100   
4350  Computer, Automated Teller, and Office Machine...            60   

         a_mean a_pct10 a_pct25 a_median a_pct75 a_pct90  year  
0         70170   41340   52120    67290   83570  113550  2000  
1         59940   40670   47920    57160   69110   86170  2000  
2         56160   37040   46420    54520   66380   79610  2000  
3         59870   42260   49110    59280   69840   82100  2000  
4         39790   24840   29060    37170   44100   54160  2000  
...         ...     ...     ...      ...     ...     ...   ...  
4346      42510   25730   31410    41430   52910   63640  2000  
4347      35570   26470   29390    32990   41090   51340  2000  
4348      24900   17120   20980    25010   28180   33150  2000  
4349      20230   12480   14810    16900   21370   38670  2000  
4350      47610   23830   29640    44140   66340   71250  2000  

[4351 rows x 13 columns]

### Clean up unhelpful values in the data set.

(* = indicates that a wage estimate is not available" - replace with 0)

(** = indicates that an employment estimate is not available - replace with 0)

(# = indicates a wage greater than $70.00 per hour or $145,600 per year" - replace with $145,601)

In [9]:
bad = ['*', '**', '#']
good = [0, 0, 145601]

In [10]:
commsa2000 = commsa2000.replace(to_replace=bad, value=good)

In [11]:
commsa2000


prim_state  area          area_name occ_code  \
0            AK   380  Anchorage, AK MSA  11-3021   
1            AK   380  Anchorage, AK MSA  13-1111   
2            AK   380  Anchorage, AK MSA  15-1021   
3            AK   380  Anchorage, AK MSA  15-1031   
4            AK   380  Anchorage, AK MSA  15-1041   
...         ...   ...                ...      ...   
4346         WY  1580   Cheyenne, WY MSA  15-1051   
4347         WY  1580   Cheyenne, WY MSA  15-1071   
4348         WY  1580   Cheyenne, WY MSA  43-9011   
4349         WY  1580   Cheyenne, WY MSA  43-9021   
4350         WY  1580   Cheyenne, WY MSA  49-2011   

                                              occ_title       tot_emp  \
0             Computer and Information Systems Managers           210   
1                                   Management Analysts           180   
2                                  Computer Programmers           380   
3             Computer Software Engineers, Applications            50   
4                          Computer Support Specialists           720   
...                                                 ...           ...   
4346                          Computer Systems Analysts            70   
4347        Network and Computer Systems Administrators            40   
4348                                 Computer Operators            80   
4349                                  Data Entry Keyers           100   
4350  Computer, Automated Teller, and Office Machine...            60   

         a_mean a_pct10 a_pct25 a_median a_pct75 a_pct90  year  
0         70170   41340   52120    67290   83570  113550  2000  
1         59940   40670   47920    57160   69110   86170  2000  
2         56160   37040   46420    54520   66380   79610  2000  
3         59870   42260   49110    59280   69840   82100  2000  
4         39790   24840   29060    37170   44100   54160  2000  
...         ...     ...     ...      ...     ...     ...   ...  
4346      42510   25730   31410    41430   52910   63640  2000  
4347      35570   26470   29390    32990   41090   51340  2000  
4348      24900   17120   20980    25010   28180   33150  2000  
4349      20230   12480   14810    16900   21370   38670  2000  
4350      47610   23830   29640    44140   66340   71250  2000  

[4351 rows x 13 columns]

### Clean out "MSA" from area_name column. Need values that Tableau can use to create a map.

In [12]:
commsa2000['area_name'] = commsa2000['area_name'].str.replace("MSA", "")

In [13]:
assert commsa2000['area_name'].str.contains('MSA').any() == False

In [14]:
commsa2000.head()

prim_state  area       area_name occ_code  \
0         AK   380  Anchorage, AK   11-3021   
1         AK   380  Anchorage, AK   13-1111   
2         AK   380  Anchorage, AK   15-1021   
3         AK   380  Anchorage, AK   15-1031   
4         AK   380  Anchorage, AK   15-1041   

                                   occ_title       tot_emp     a_mean a_pct10  \
0  Computer and Information Systems Managers           210      70170   41340   
1                        Management Analysts           180      59940   40670   
2                       Computer Programmers           380      56160   37040   
3  Computer Software Engineers, Applications            50      59870   42260   
4               Computer Support Specialists           720      39790   24840   

  a_pct25 a_median a_pct75 a_pct90  year  
0   52120    67290   83570  113550  2000  
1   47920    57160   69110   86170  2000  
2   46420    54520   66380   79610  2000  
3   49110    59280   69840   82100  2000  
4   29060    37170   44100   54160  2000

In [15]:
import re

### Use a regular expression and str.replace to clean the space P off the end of some area_name values

In [16]:
commsa2000['area_name'] = commsa2000['area_name'].str.replace(r' P$', '', regex=True)
print (commsa2000)

     prim_state  area       area_name occ_code  \
0            AK   380  Anchorage, AK   11-3021   
1            AK   380  Anchorage, AK   13-1111   
2            AK   380  Anchorage, AK   15-1021   
3            AK   380  Anchorage, AK   15-1031   
4            AK   380  Anchorage, AK   15-1041   
...         ...   ...             ...      ...   
4346         WY  1580   Cheyenne, WY   15-1051   
4347         WY  1580   Cheyenne, WY   15-1071   
4348         WY  1580   Cheyenne, WY   43-9011   
4349         WY  1580   Cheyenne, WY   43-9021   
4350         WY  1580   Cheyenne, WY   49-2011   

                                              occ_title       tot_emp  \
0             Computer and Information Systems Managers           210   
1                                   Management Analysts           180   
2                                  Computer Programmers           380   
3             Computer Software Engineers, Applications            50   
4                          Compute

#### Checkpoint: The year 2000 should have 4351 relevant records, based on OCC codes of interest.

### Read in 2010 MSA OEWS csv files.

In [17]:
msa2010_1 = pd.read_csv('../data/OEWS/2010MSA_M_dl_1.csv')
msa2010_1.head(2)

PRIM_STATE   AREA      AREA_NAME OCC_CODE               OCC_TITLE  GROUP  \
0         AK  11260  Anchorage, AK  00-0000         All Occupations  total   
1         AK  11260  Anchorage, AK  11-0000  Management Occupations  major   

   TOT_EMP EMP_PRSE JOBS_1000 LOC QUOTIENT  ... H_MEDIAN H_PCT75 H_PCT90  \
0  164,770      1.4  1000.000        1.000  ...    20.11   31.48   43.89   
1   10,720      2.4    65.082        1.373  ...    40.12   54.88   73.54   

  A_PCT10 A_PCT25 A_MEDIAN  A_PCT75  A_PCT90 ANNUAL HOURLY  
0  20,420  28,040   41,830   65,480   91,280    NaN    NaN  
1  44,630  60,500   83,440  114,150  152,960    NaN    NaN  

[2 rows x 25 columns]

In [18]:
msa2010_2 = pd.read_csv('../data/OEWS/2010MSA_M_dl_2.csv')
msa2010_2.head(2)

PRIM_STATE   AREA     AREA_NAME OCC_CODE               OCC_TITLE  GROUP  \
0         KS  29940  Lawrence, KS  00-0000         All Occupations  total   
1         KS  29940  Lawrence, KS  11-0000  Management Occupations  major   

  TOT_EMP EMP_PRSE JOBS_1000 LOC QUOTIENT  ... H_MEDIAN H_PCT75 H_PCT90  \
0  45,880      9.7  1000.000        1.000  ...    13.31   21.23   31.68   
1   2,090     19.8    45.520        0.961  ...    33.89   46.87   65.72   

  A_PCT10 A_PCT25 A_MEDIAN A_PCT75  A_PCT90 ANNUAL HOURLY  
0  16,340  18,880   27,680  44,150   65,900    NaN    NaN  
1  37,550  48,230   70,480  97,490  136,690    NaN    NaN  

[2 rows x 25 columns]

In [19]:
msa2010_3 = pd.read_csv('../data/OEWS/2010MSA_M_dl_3.csv')
msa2010_3.head(2)

PRIM_STATE   AREA  AREA_NAME OCC_CODE               OCC_TITLE  GROUP  \
0         OH  10420  Akron, OH  00-0000         All Occupations  total   
1         OH  10420  Akron, OH  11-0000  Management Occupations  major   

   TOT_EMP EMP_PRSE JOBS_1000 LOC QUOTIENT  ... H_MEDIAN H_PCT75 H_PCT90  \
0  307,140      1.8  1000.000        1.000  ...    15.27   24.10   35.29   
1   12,360      2.8    40.239        0.849  ...    43.52   60.77       #   

  A_PCT10 A_PCT25 A_MEDIAN  A_PCT75 A_PCT90 ANNUAL HOURLY  
0  16,930  20,680   31,760   50,120  73,410    NaN    NaN  
1  46,920  65,510   90,530  126,400       #    NaN    NaN  

[2 rows x 25 columns]

### Create the 2010 MSA dataframe

In [20]:
commsa2010 = msa2010_1.append([msa2010_2, msa2010_3], ignore_index=True, verify_integrity=True, sort=False)
commsa2010.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 149428 entries, 0 to 149427
Data columns (total 25 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   PRIM_STATE    149428 non-null  object
 1   AREA          149428 non-null  int64 
 2   AREA_NAME     149428 non-null  object
 3   OCC_CODE      149428 non-null  object
 4   OCC_TITLE     149428 non-null  object
 5   GROUP         9169 non-null    object
 6   TOT_EMP       149428 non-null  object
 7   EMP_PRSE      149428 non-null  object
 8   JOBS_1000     149428 non-null  object
 9   LOC QUOTIENT  149428 non-null  object
 10  H_MEAN        149428 non-null  object
 11  A_MEAN        149428 non-null  object
 12  MEAN_PRSE     149428 non-null  object
 13  H_PCT10       149428 non-null  object
 14  H_PCT25       149428 non-null  object
 15  H_MEDIAN      149428 non-null  object
 16  H_PCT75       149428 non-null  object
 17  H_PCT90       149428 non-null  object
 18  A_PCT10       149428 non

#### Select only the columns you need. Add a year column equal to 2010.

In [21]:
commsa2010 = commsa2010[['PRIM_STATE','AREA', 'AREA_NAME', 'OCC_CODE', 'OCC_TITLE', 'TOT_EMP', 'JOBS_1000', 'LOC QUOTIENT',
                         'A_MEAN', 'A_PCT10', 'A_PCT25', 'A_MEDIAN', 'A_PCT75', 'A_PCT90']]
commsa2010['year'] = 2010
commsa2010.head()

PRIM_STATE   AREA      AREA_NAME OCC_CODE  \
0         AK  11260  Anchorage, AK  00-0000   
1         AK  11260  Anchorage, AK  11-0000   
2         AK  11260  Anchorage, AK  11-1011   
3         AK  11260  Anchorage, AK  11-1021   
4         AK  11260  Anchorage, AK  11-2011   

                             OCC_TITLE  TOT_EMP JOBS_1000 LOC QUOTIENT  \
0                      All Occupations  164,770  1000.000        1.000   
1               Management Occupations   10,720    65.082        1.373   
2                     Chief Executives      530     3.237        1.504   
3      General and Operations Managers    4,100    24.892        1.852   
4  Advertising and Promotions Managers       50     0.333        1.313   

    A_MEAN A_PCT10  A_PCT25 A_MEDIAN  A_PCT75  A_PCT90  year  
0   51,470  20,420   28,040   41,830   65,480   91,280  2010  
1   94,110  44,630   60,500   83,440  114,150  152,960  2010  
2  157,130  82,960  105,870  137,190        #        #  2010  
3   93,340  43,250   58,920   81,420  115,300  156,430  2010  
4   66,660  41,370   45,520   66,740   85,320  100,500  2010

### Use a string method to change the case of the column headers.

In [22]:
col2010 = pd.Series(['PRIM_STATE','AREA', 'AREA_NAME', 'OCC_CODE', 'OCC_TITLE', 'TOT_EMP', 'JOBS_1000', 'LOC QUOTIENT', 'A_MEAN', 
                         'A_PCT10', 'A_PCT25', 'A_MEDIAN', 'A_PCT75', 'A_PCT90', 'year'])
lowcol2010 = col2010.str.lower()
print(lowcol2010)
commsa2010.columns = lowcol2010
commsa2010.head()

0       prim_state
1             area
2        area_name
3         occ_code
4        occ_title
5          tot_emp
6        jobs_1000
7     loc quotient
8           a_mean
9          a_pct10
10         a_pct25
11        a_median
12         a_pct75
13         a_pct90
14            year
dtype: object


prim_state   area      area_name occ_code  \
0         AK  11260  Anchorage, AK  00-0000   
1         AK  11260  Anchorage, AK  11-0000   
2         AK  11260  Anchorage, AK  11-1011   
3         AK  11260  Anchorage, AK  11-1021   
4         AK  11260  Anchorage, AK  11-2011   

                             occ_title  tot_emp jobs_1000 loc quotient  \
0                      All Occupations  164,770  1000.000        1.000   
1               Management Occupations   10,720    65.082        1.373   
2                     Chief Executives      530     3.237        1.504   
3      General and Operations Managers    4,100    24.892        1.852   
4  Advertising and Promotions Managers       50     0.333        1.313   

    a_mean a_pct10  a_pct25 a_median  a_pct75  a_pct90  year  
0   51,470  20,420   28,040   41,830   65,480   91,280  2010  
1   94,110  44,630   60,500   83,440  114,150  152,960  2010  
2  157,130  82,960  105,870  137,190        #        #  2010  
3   93,340  43,250   58,920   81,420  115,300  156,430  2010  
4   66,660  41,370   45,520   66,740   85,320  100,500  2010

In [23]:
bad1 = ['*', '**', '#']
good1 = [0, 0, 145601]
commsa2010 = commsa2010.replace(to_replace=bad1, value=good1)
commsa2010['area_name'] = commsa2010['area_name'].str.replace(r' Metropolitan Division$', '', regex=True)
commsa2010['area_name'] = commsa2010['area_name'].str.replace(r' NECTA Division$', '', regex=True)
commsa2010.head(4)

prim_state   area      area_name occ_code                        occ_title  \
0         AK  11260  Anchorage, AK  00-0000                  All Occupations   
1         AK  11260  Anchorage, AK  11-0000           Management Occupations   
2         AK  11260  Anchorage, AK  11-1011                 Chief Executives   
3         AK  11260  Anchorage, AK  11-1021  General and Operations Managers   

   tot_emp jobs_1000 loc quotient   a_mean a_pct10  a_pct25 a_median  a_pct75  \
0  164,770  1000.000        1.000   51,470  20,420   28,040   41,830   65,480   
1   10,720    65.082        1.373   94,110  44,630   60,500   83,440  114,150   
2      530     3.237        1.504  157,130  82,960  105,870  137,190   145601   
3    4,100    24.892        1.852   93,340  43,250   58,920   81,420  115,300   

   a_pct90  year  
0   91,280  2010  
1  152,960  2010  
2   145601  2010  
3  156,430  2010

### Keep only relevant OCC codes for the year 2010.

In [24]:
occ_codes_2010 = ['11-3021', '13-1081', '13-1111', '13-1161', '15-1111', '15-1121', '15-1122', '15-1131', '15-1132', '15-1133',
                '15-1134', '15-1141', '15-1142', '15-1143', '15-1150','15-1151', '15-1152', '15-1179', '15-1199', '15-1799',
                  '15-2031', '15-2041', '17-2061', '43-9011', '43-9021', '43-9111', '49-2011']
commsa2010 = commsa2010.loc[commsa2010.occ_code.isin(occ_codes_2010)].reset_index(drop=True)

print(commsa2010.shape)
commsa2010.tail()


(5486, 15)


prim_state   area     area_name occ_code  \
5481         WY  16940  Cheyenne, WY  15-1142   
5482         WY  16940  Cheyenne, WY  15-1150   
5483         WY  16940  Cheyenne, WY  15-1799   
5484         WY  16940  Cheyenne, WY  43-9021   
5485         WY  16940  Cheyenne, WY  49-2011   

                                              occ_title tot_emp jobs_1000  \
5481       Network and Computer Systems Administrators*      80     2.003   
5482                       Computer Support Specialists     180     4.499   
5483                   Computer Occupations, All Other*      80     1.991   
5484                                  Data Entry Keyers      60     1.499   
5485  Computer, Automated Teller, and Office Machine...      50     1.267   

     loc quotient  a_mean a_pct10 a_pct25 a_median a_pct75 a_pct90  year  
5481        0.764  54,000  41,300  44,840   52,050  59,460  71,030  2010  
5482        0.987  48,420  37,110  41,290   46,610  54,360  61,690  2010  
5483        1.382  63,700  32,130  54,030   65,290  75,690  84,850  2010  
5484        0.868  19,570  15,910  16,950   18,590  21,650  24,830  2010  
5485        1.460  32,440  17,900  23,450   35,360  39,720  43,090  2010

#### Checkpoint: The year 2010 should have 5486 rows, based on OCC codes of interest. (Before catching & adding 15-1799 it was 5234 rows).


### Append 2010 dataframe onto 2000 dataframe.

In [25]:
combmsa2000_2010 = commsa2000.append(commsa2010, ignore_index=True, verify_integrity=True, sort=False)
combmsa2000_2010.info()
# Go back and fix a_wpct10 and a_pct10 column name differences. - Done 8/9.
# Go back and fix occ_titl to occ_title in year 2000 so it matches up with 2010. - Done 8/10.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9837 entries, 0 to 9836
Data columns (total 15 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   prim_state    9837 non-null   object
 1   area          9837 non-null   int64 
 2   area_name     9837 non-null   object
 3   occ_code      9837 non-null   object
 4   occ_title     9837 non-null   object
 5   tot_emp       9837 non-null   object
 6   a_mean        9837 non-null   object
 7   a_pct10       9837 non-null   object
 8   a_pct25       9837 non-null   object
 9   a_median      9837 non-null   object
 10  a_pct75       9837 non-null   object
 11  a_pct90       9837 non-null   object
 12  year          9837 non-null   int64 
 13  jobs_1000     5486 non-null   object
 14  loc quotient  5486 non-null   object
dtypes: int64(2), object(13)
memory usage: 1.1+ MB


### Read in 2017 OEWS MSA data.

In [26]:
msa2017 = pd.read_csv('../data/OEWS/2017MSA_M_dl.csv')
msa2017.head(2)

PRIM_STATE   AREA      AREA_NAME OCC_CODE               OCC_TITLE OCC_GROUP  \
0         AK  11260  Anchorage, AK  00-0000         All Occupations     total   
1         AK  11260  Anchorage, AK  11-0000  Management Occupations     major   

   TOT_EMP EMP_PRSE JOBS_1000 LOC QUOTIENT  ... H_MEDIAN H_PCT75 H_PCT90  \
0  173,420      1.3  1000.000         1.00  ...    22.76   35.80   50.85   
1    9,950      2.5    57.380         1.12  ...    48.82   65.62   92.45   

  A_PCT10 A_PCT25 A_MEDIAN  A_PCT75  A_PCT90 ANNUAL HOURLY  
0  23,170  30,820   47,340   74,470  105,760    NaN    NaN  
1  53,570  74,470  101,560  136,490  192,300    NaN    NaN  

[2 rows x 25 columns]

In [27]:
msa2017 = msa2017[['PRIM_STATE','AREA', 'AREA_NAME', 'OCC_CODE', 'OCC_TITLE', 'TOT_EMP', 'JOBS_1000', 'LOC QUOTIENT', 'A_MEAN', 
                         'A_PCT10', 'A_PCT25', 'A_MEDIAN', 'A_PCT75', 'A_PCT90']]
msa2017['year'] = 2017
bad1 = ['*', '**', '#']
good1 = [0, 0, 145601]
msa2017 = msa2017.replace(to_replace=bad1, value=good1)
msa2017['AREA_NAME'] = msa2017['AREA_NAME'].str.replace(r' Metropolitan Division$', '', regex=True)
msa2017['AREA_NAME'] = msa2017['AREA_NAME'].str.replace(r' NECTA Division$', '', regex=True)
msa2017.head()

PRIM_STATE   AREA      AREA_NAME OCC_CODE  \
0         AK  11260  Anchorage, AK  00-0000   
1         AK  11260  Anchorage, AK  11-0000   
2         AK  11260  Anchorage, AK  11-1011   
3         AK  11260  Anchorage, AK  11-1021   
4         AK  11260  Anchorage, AK  11-2011   

                             OCC_TITLE  TOT_EMP JOBS_1000 LOC QUOTIENT  \
0                      All Occupations  173,420  1000.000         1.00   
1               Management Occupations    9,950    57.380         1.12   
2                     Chief Executives      390     2.235         1.52   
3      General and Operations Managers    3,450    19.882         1.28   
4  Advertising and Promotions Managers       50     0.265         1.34   

    A_MEAN A_PCT10  A_PCT25 A_MEDIAN  A_PCT75  A_PCT90  year  
0   58,980  23,170   30,820   47,340   74,470  105,760  2017  
1  114,980  53,570   74,470  101,560  136,490  192,300  2017  
2  176,890  86,480  116,870  157,420   145601   145601  2017  
3  116,770  52,670   72,230   99,150  142,440  201,120  2017  
4  104,210  54,600   64,700   80,780  140,520  192,860  2017

In [28]:
occ_codes_2017 = ['11-3021', '13-1081', '13-1111', '13-1161', '15-1111', '15-1121', '15-1122', '15-1131', '15-1132', '15-1133',
                '15-1134', '15-1141', '15-1142', '15-1143', '15-1151', '15-1152', '15-1199', '15-2031', '15-2041', '17-2061',
                '43-9011', '43-9021', '43-9111', '49-2011']
msa2017 = msa2017.loc[msa2017.OCC_CODE.isin(occ_codes_2017)].reset_index(drop=True)

print(msa2017.shape)
msa2017.tail()

(6552, 15)


PRIM_STATE   AREA     AREA_NAME OCC_CODE  \
6547         WY  16940  Cheyenne, WY  15-1141   
6548         WY  16940  Cheyenne, WY  15-1142   
6549         WY  16940  Cheyenne, WY  15-1151   
6550         WY  16940  Cheyenne, WY  15-1152   
6551         WY  16940  Cheyenne, WY  43-9021   

                                        OCC_TITLE TOT_EMP JOBS_1000  \
6547                      Database Administrators     130     2.863   
6548  Network and Computer Systems Administrators     120     2.688   
6549            Computer User Support Specialists     190     4.085   
6550         Computer Network Support Specialists      60     1.388   
6551                            Data Entry Keyers       0         0   

     LOC QUOTIENT  A_MEAN A_PCT10 A_PCT25 A_MEDIAN A_PCT75 A_PCT90  year  
6547         3.59  60,460  45,150  47,190   57,220  69,860  84,080  2017  
6548         1.02  65,700  40,300  50,460   63,610  77,550  99,530  2017  
6549         0.95  54,240  36,180  43,720   52,900  60,770  76,240  2017  
6550         1.06  64,630  31,780  54,970   64,860  74,610  81,320  2017  
6551            0  23,140  17,330  19,090   22,330  27,260  30,720  2017

In [29]:
col2017 = pd.Series(['PRIM_STATE','AREA', 'AREA_NAME', 'OCC_CODE', 'OCC_TITLE', 'TOT_EMP', 'JOBS_1000', 'LOC QUOTIENT',
                     'A_MEAN', 'A_PCT10', 'A_PCT25', 'A_MEDIAN', 'A_PCT75', 'A_PCT90', 'year'])
lowcol2017 = col2017.str.lower()
print(lowcol2017)
msa2017.columns = lowcol2017
msa2017.head()

0       prim_state
1             area
2        area_name
3         occ_code
4        occ_title
5          tot_emp
6        jobs_1000
7     loc quotient
8           a_mean
9          a_pct10
10         a_pct25
11        a_median
12         a_pct75
13         a_pct90
14            year
dtype: object


prim_state   area      area_name occ_code  \
0         AK  11260  Anchorage, AK  11-3021   
1         AK  11260  Anchorage, AK  13-1081   
2         AK  11260  Anchorage, AK  13-1111   
3         AK  11260  Anchorage, AK  13-1161   
4         AK  11260  Anchorage, AK  15-1121   

                                           occ_title tot_emp jobs_1000  \
0          Computer and Information Systems Managers     220     1.288   
1                                       Logisticians     200     1.182   
2                                Management Analysts     430     2.503   
3  Market Research Analysts and Marketing Special...     350     2.013   
4                          Computer Systems Analysts     610     3.501   

  loc quotient   a_mean a_pct10 a_pct25 a_median  a_pct75  a_pct90  year  
0         0.50  110,000  74,080  90,230  108,890  127,170  151,680  2017  
1         1.05   88,010  60,480  70,960   82,390  102,360  124,040  2017  
2         0.54   87,930  55,700  68,730   85,070  101,140  123,010  2017  
3         0.48   70,930  37,120  48,850   67,490   92,940  116,630  2017  
4         0.86   80,110  44,320  60,750   80,520   97,970  117,080  2017

In [30]:
combmsa00_10_17 = combmsa2000_2010.append(msa2017, ignore_index=True, verify_integrity=True, sort=False)
combmsa00_10_17.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16389 entries, 0 to 16388
Data columns (total 15 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   prim_state    16389 non-null  object
 1   area          16389 non-null  int64 
 2   area_name     16389 non-null  object
 3   occ_code      16389 non-null  object
 4   occ_title     16389 non-null  object
 5   tot_emp       16389 non-null  object
 6   a_mean        16389 non-null  object
 7   a_pct10       16389 non-null  object
 8   a_pct25       16389 non-null  object
 9   a_median      16389 non-null  object
 10  a_pct75       16389 non-null  object
 11  a_pct90       16389 non-null  object
 12  year          16389 non-null  int64 
 13  jobs_1000     12038 non-null  object
 14  loc quotient  12038 non-null  object
dtypes: int64(2), object(13)
memory usage: 1.9+ MB


In [31]:
#new = combmsa00_10_17['area_name'].str.split("-|,", n = 3, expand = True)combmsa00_10_17['clean_area_name']= new[0]
# msa2017['AREA_NAME'] = msa2017['AREA_NAME'].str.replace(r' Metropolitan Division$', '', regex=True)

## Create master csv to feed into Tableau 

In [32]:
combmsa00_10_17.to_csv('../data/OEWS/2000_10_17_msacombined.csv')

### Determine the total number of records affected by replacing # with $145,601. % of records are affected.

In [33]:
mask90th = commsa2000.a_pct90 == 145601
print(commsa2000[mask90th].shape)
mask75th = commsa2000.a_pct75 == 145601
print(commsa2000[mask75th].shape)


(50, 13)
(8, 13)


In [34]:
mask90th = commsa2010.a_pct90 == 145601
print(commsa2010[mask90th].shape)
mask75th = commsa2010.a_pct75 == 145601
print(commsa2010[mask75th].shape)
mask50th = commsa2010.a_median == 145601
print(commsa2010[mask50th].shape)


(175, 15)
(15, 15)
(2, 15)


In [35]:
mask90th = msa2017.a_pct90 == 145601
print(msa2017[mask90th].shape)
mask75th = msa2017.a_pct75 == 145601
print(msa2017[mask75th].shape)
mask50th = msa2017.a_median == 145601
print(msa2017[mask50th].shape)

(106, 15)
(6, 15)
(0, 15)


### Pause here and see how well Tableau map capabilities can handle the contents of the area_name column.
Results - floating P's at the end of some MSA area_name values